docs/ECOS.md

```diff
@@ -0,0 +1,141 @@
+
+EVE TSC plans to submit ECO and ECI standards for broad Linux Foundation adoption once this document reaches the state of a formal specification. At the current stage, however, we are still collecting product requirements and recording them (together with implementation details) below.
+
+Working with ECOs will typically go through multiple stages, each with its own expectations around standardization and how they interact with the rest of the cloud-native developer and DevOps toolchains. We expect EVE users to...
+
+   #1 ...build an Edge Container Image (ECI) using the usual tools typically available on developer's desktop. The final build artifact may either be hidden in a local cache of some kind (along the lines of Docker image repository, local Maven cache, etc.) or be immediately available as a self-contained binary file
+   #2 ...instantiate ECOs based on a given ECI in a simulated environment that can easily run on developer's desktop (unlike a real EVE instance that typically runs on an Edge Node). Input to this stage could either be a reference to an ECI hidden in a local cache or a URL (including a local URL) for a self-contained binary file
+   #3 ...share ECIs with other developers
+   #4 ...deploy ECI on a fleet of Edge Nodes running EVE
+   #5 ...manage a rich life-cycle of an ECO (creation, stopping, re-starting, etc.)
+
+In the following sections we define semantics of all these actions and expectations around how they dovetail into each other.
+
+## ECO Runtime Specification
+
+ECO Runtime Specification aims to specify the binding of ECI to Edge Node's hardware and software resources, configuration, execution environment, and lifecycle of an Edge Container. Note that unlike most of the runtimes covered by OCI specification, EVE is a self-contained execution engine. This allows us to skip a lot of details (around on-disk container runtime formats, configuration schema, etc.) that otherwise would have to be standardized.
+
+Since there is always a 1-many mapping between ECI and ECOs, a very important question is how do you reference a seed ECI for any ECO operation requiring it (like creating an ECO from an ECI). There are two methods that are covered in the following section: you can reference an immutable ECI via a content-addressable handle or you can reference an ECI via a human readable name that maps to an immutable one. Since the later kind of mapping can be updated at will, it is very important to understand how various transitions in the ECO state transition diagram react to an ECI mapping changing while transition is in flight.
+
+Since nothing happens with EVE unless initiated by a Controller through a certain set of configuration directives, this section focuses on:
+
```

Comment:
Ordered list formatting needs to convert to markdown.

Reply:
fixed